### ISSUE TITLE
Purpose/Goals of Repo

### ISSUE THREAD

**Issue by djrtwo** [RigoBlock/SolidityCasper#4]
Hey! I just wanted to get a sense of your goals with this repository.

Are you looking port the full viper implementation to Solidity?
I've been considering doing so and to write a unit-test suite to test against both solidity and viper implementations. This interface is obviously a great place to start.

If you're interested in going down that road, we should discuss further and choose how we want to implement tests.

Cheers!

**Comment by gabririgo** [comment]
Hey Danny (I amended a bit as I initially thought it was an email..!), 
First of all thank you.
My idea was to create the interface, write the documentation for the calls and, after that, start work on the javascript library for making it a node module for developers, and testing of course.
Writing casper in solidity was not in my first intentions, as I am mostly interested in interacting with casper. 
However, I am happy to coordinate the complete traduction and work with you on it as it seems the project could be of use to many, especially users who are waiting for the casper fork to be finalized. I had budgeted tests for casper, which have not started yet, hence I can save some time for working on the single functions. I use condition oriented programming (formalized by parity) for solidity styling.
Guidelines for casper tests are in LLL but this part is not 100% clear to me.
Happy to coordinate on this, a good chance to provide a useful open source project.

Cheers!

**Comment by djrtwo** [comment]
Got it.

Keep me updated. I'll happily do code review on the interface PRs, and might push forward on porting the viper implementation to solidity. Will let you know if I end up having time on that.

In the future for changes like https://github.com/RigoBlock/SolidityCasper/commit/d38f32a555628cbb8211fa279b6b481f79947ed5, if you open up a Pull Request, I'll happily do code review before it enters master.

**Comment by gabririgo** [comment]
yes that's a very good idea. Casper should be finalized in the next few days/weeks as tests on ropsten are expected, as soon as it's finalized i'll run some tests and update the repo again

**Comment by gabririgo** [comment]
Hi Danny, I see you are furiously active on the casper docs and review! :)
I just updated the casper interface contract and casper interface, created a pull request from the branch latest-casper-march. Would love to merge and delete the branch, please let me know if you can review it before committing to master. Thank you

**Comment by djrtwo** [comment]
Absolutely! Will get to it in next 24 hours.
